New Resource: `azurerm_spring_cloud_service` (#4928)
terraform-providers/terraform-provider-azurerm: website/docs/r/spring_cloud.html.markdown

subcategory: "App Platform"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_spring_cloud"
sidebar_current: "docs-azurerm-resource-spring-cloud"
description: |-

>   Manage Azure Spring Cloud Service instance.

Reviewer: ```suggestion
  Manages an Azure Spring Cloud Service.
```

---

# azurerm_spring_cloud

> Manage Azure Spring Cloud Service instance. Azure Spring Cloud provides a managed service that enables Java developers to easily build and run Spring-boot based microservices on Azure with no code changes.

Reviewer: ```suggestion
Manages an Azure Spring Cloud Service. Azure Spring Cloud provides a managed service that enables Java developers to easily build and run Spring-boot based microservices on Azure with no code changes.
```

Within an azure spring cloud service, users can manage config server, manage multiple spring cloud apps and manage deployments

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {

>   name     = "resGroup"

Reviewer: ```suggestion
  name     = "example-resources"
```

  location = "Southeast Asia"
}

resource "azurerm_spring_cloud" "example" {
  name                     = "sc-example"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location

  tags = {
    environment = "staging"
  }
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) The name of the Service resource. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal. Changing this forces a new resource to be created.

> * `location` - (Optional) The GEO location of the resource. Changing this forces a new resource to be created. Until November 20, 2019 4:20 PM, only `East US`, `Southeast Asia`, `West Europe`, `West US 2` following locations are supported.

Reviewer: We are past this date, is this still true?

Author: yes, it still only support these four locations.
In the azure portal, it's the same

* `tags` - (Optional) Tags of the service which is a list of key value pairs that describe the resource. Changing this forces a new resource to be created.

## Attributes Reference

The following attributes are exported:

* `version` - Version of the Service

* `service_id` - ServiceInstanceEntity GUID which uniquely identifies a created resource

* `id` - Fully qualified resource Id for the resource.

> * `name` - The name of the resource.

Reviewer: COuld we make the wording here consistent with the rest of the docs?

---
subcategory: "App Platform"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_spring_cloud"
description: |-
  Manages an Azure Spring Cloud Service.
---

# azurerm_spring_cloud

Manages an Azure Spring Cloud Service.

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "Southeast Asia"
}

resource "azurerm_spring_cloud" "example" {

>   name                = "sc-example"

Reviewer 2: for examples we use `example-{name}`

```suggestion
  name                = "example-springcloud"
```

  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  tags = {
    Env = "staging"
  }
}
```

## Argument Reference

The following arguments are supported:

> * `name` - (Required) The name of the Service resource. Changing this forces a new resource to be created.

Reviewer 2: can we update this to match the terminology used by other resources?

> * `resource_group_name` - (Required) The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal. Changing this forces a new resource to be created.

Reviewer 2: can we update this to match the terminology used by other resources?

> * `location` - (Required) The GEO location of the resource. Changing this forces a new resource to be created. Until November 20, 2019 4:20 PM, only `East US`, `Southeast Asia`, `West Europe`, `West US 2` following locations are supported.

Reviewer 2: can we update this to match the terminology used by other resources here?

the "only supported in X regions" can be achieved using an info box:

```
-> **Note:** At this time Azure Spring Cloud is only supported in a subset of regions (including `East US`, `South East Asia`, `West Europe` and `West US 2`.
```

> * `tags` - (Optional) Tags of the service which is a list of key value pairs that describe the resource. Changing this forces a new resource to be created.

Reviewer 2: can we update this to use the same terminology as the other resources here?

## Attributes Reference

The following attributes are exported:

> * `id` - The id of the Spring Cloud Service.

Reviewer 2: `id` wants capitalizing here:

```suggestion
* `id` - The ID of the Spring Cloud Service.
```

> * `version` - The version of the Spring Cloud Service.

Reviewer 2: this can be removed from the documentation since this field isn't worth exposing at this point in time

> * `service_id` - The GUID of the Spring Cloud Service which uniquely identifies.

Reviewer 2: this can be removed from the documentation since this field isn't worth exposing at this point in time
